### PR TITLE
Factor out pluralisation into syntax::errors

### LINKED ISSUE #64238: Factor out pluralisation checks in diagnostics
The following pattern is common in diagnostic messages in rustc:

```rust
format!("found {} thing{}", x, if x != 1 { "s" } else { "" })
```

It would be good to extract all of these pluralisation checks into a function or macro:

```rust
format!("found {} thing{}", x, pluralise(x))
```

There's already one in src/librustc/ty/error.rs, but it's not used anywhere else, so we could move it to somewhere in src/librustc_errors, and replace the occurrences of this pattern with the macro.

https://github.com/rust-lang/rust/blob/0b97726e6c524d2cc0de4c2f5b1284eca010a7b2/src/librustc/ty/error.rs#L85-L89

This issue has been assigned to @V1shvesh via [this comment](https://github.com/rust-lang/rust/issues/64238#issuecomment-529086433).

### PR DESCRIPTION
**Pull request by V1shvesh**
Fixes #64238.

### CODE REVIEW

File: src/librustc_errors/lib.rs [hunk 1]
```diff
@@ -845,3 +845,11 @@ impl Level {
         }
     }
 }
+
+#[macro_export]
+macro_rules! pluralise {
+            ($x:expr) => {
```
Comment (Centril) [inline src/librustc_errors/lib.rs]: Dedent this. =P

Comment (V1shvesh) [reply src/librustc_errors/lib.rs]: Ahh, missed that. Thanks @Centril!
Fixed in commit 7457ef8. :)

**Comment by Centril** [comment]
Beautiful 👍 Let's land this.

r? @Centril @bors r+ rollup

**Comment by V1shvesh** [comment]
And there goes my first contribution towards Rust! The first of many, I hope... :stuck_out_tongue: 

@Centril @varkor Thank you so much for the help! :smile:

**Pull request merged**